lambda/custom/index.js

```diff
@@ -148,24 +167,80 @@ const PostEmojiMessageIntentHandler = {
 
 const GetLastMessageFromChannelIntentHandler = {
 	canHandle(handlerInput) {
-		return handlerInput.requestEnvelope.request.type === 'IntentRequest'
-      && handlerInput.requestEnvelope.request.intent.name === 'GetLastMessageFromChannelIntent';
+		return handlerInput.requestEnvelope.request.type === 'IntentRequest' &&
+			handlerInput.requestEnvelope.request.intent.name === 'GetLastMessageFromChannelIntent';
 	},
 	async handle(handlerInput) {
 		try {
-			const { accessToken } = handlerInput.requestEnvelope.context.System.user;
+			const {
+				accessToken
+			} = handlerInput.requestEnvelope.context.System.user;
 
 			const channelNameData = handlerInput.requestEnvelope.request.intent.slots.getmessagechannelname.value;
 			const channelName = helperFunctions.replaceWhitespacesFunc(channelNameData);
 
 			const headers = await helperFunctions.login(accessToken);
+			const fileurl = await helperFunctions.getLastMessageFileURL(channelName, headers);
+			const download = await helperFunctions.getLastMessageFileDowloadURL(fileurl, headers);
 			const speechText = await helperFunctions.channelLastMessage(channelName, headers);
 
+
+		if(supportsAPL(handlerInput)){
+
 			return handlerInput.jrb
 				.speak(speechText)
 				.reprompt(speechText)
-				.withSimpleCard(ri('GET_LAST_MESSAGE_FROM_CHANNEL.CARD_TITLE'), speechText)
+				.addDirective({
+					type: 'Alexa.Presentation.APL.RenderDocument',
+					version: '1.0',
+					document: main,
+					datasources:
+
+					{
+
+						"bodyTemplate6Data": {
+							"type": "object",
+							"objectId": "bt6Sample",
+							"backgroundImage": {
+								"sources": [{
+
+										"url": download,
+										"size": "small",
+
+									},
+									{
+										"url": download,
+										"size": "large",
+
+									}
+								]
+							},
+							"textContent": {
+								"primaryText": {
+									"type": "PlainText",
+									"text": "This Is An Image Message"
+								}
+							},
+							"logoUrl": "https://github.com/RocketChat/Rocket.Chat.Artwork/raw/master/Logos/icon-circle-1024.png",
+							"hintText": "SAMPLE REDIRECTION URL TEST"
+						}
+
+
+					}
+
+
+				})
+				.getResponse();
+
+			}
+
+			else{
+				return handlerInput.jrb
+				.speak(speechText)
+				.reprompt(speechText)
 				.getResponse();
```

Comment:
Did you miss `.withSimpleCard(ri('GET_LAST_MESSAGE_FROM_CHANNEL.CARD_TITLE'), speechText)` before `getResponse()`?

Reply:
no I have added APL, which is far more superior to a simple card. So we just don't require it anymore 👍

Reply:
But it's an else statement used when APL is not supported, so it should return a card too, shouldn't it?

Reply:
@PrajvalRaval please add the simpleCard as well.
This will display in the mobile app for all devices, irrespective of whether the device supports APL or not.